### ISSUE TITLE
Obsolete members are not reported for implicit range indexer

### ISSUE THREAD

**Issue by alrz** [dotnet/roslyn#53418]
**Version Used**: main

**Steps to Reproduce**:

```cs
public class C {
    [Obsolete("error", error: true)]
    public int Slice(int i, int j) => 0;
    [Obsolete("error", error: true)]
    public int this[int i] => 0;
    [Obsolete("error", error: true)]
    public int Count => 0;
    public void M() {
        _ = this[^1]; // no error
        _ = this[..]; // no error
    }
}
```

**Expected Behavior**: Error on both usages

**Actual Behavior**: No error

**Comment by alrz** [comment]
I wanted to cover this for list patterns but that would be a breaking change.

**Comment by jcouv** [comment]
@Youssef1313 I agree this is a bug. I'm not sure whether we want to take the break. Will start thread with compat council to make decision.

**Comment by Youssef1313** [comment]
@jcouv @AlekseyTs Is it okay to reopen https://github.com/dotnet/roslyn/pull/53421 now since the untriaged label is removed?

**Comment by jcouv** [comment]
Let me confirm on Monday based on thread with compat council.

**Comment by jcouv** [comment]
Approved by compat council.

**Comment by Youssef1313** [comment]
Thanks @jcouv. Can you please reopen #53421?

**Comment by jcouv** [comment]
The "reopen" button is grayed out for me :-/
It looks like maybe the branch was deleted. But the PR still has links for the commits, so the change isn't lost.

**Comment by Youssef1313** [comment]
> The "reopen" button is grayed out for me :-/
> It looks like maybe the branch was deleted. But the PR still has links for the commits, so the change isn't lost.

Ah sorry, I forgot to restore the branch before asking a reopen 😄 
Opened a new PR anyway.

**Comment by jcouv** [comment]
Closing this issue as fixed by https://github.com/dotnet/roslyn/pull/53463
Thanks @Youssef1313
